src/utils/idl_manager.py

```
@@ -0,0 +1,348 @@
+    """Centralized manager for IDL parsers across all platforms."""
+    
+    def __init__(self):
+        """Initialize the IDL manager."""
+        self._parsers: dict[Platform, IDLParser] = {}
+        self._idl_paths: dict[Platform, str] = {}
+        self._setup_platform_idl_paths()
+    
+    def _setup_platform_idl_paths(self) -> None:
+        """Setup IDL file paths for each platform."""
+        # Get the project root directory (3 levels up from this file)
+        current_dir = os.path.dirname(os.path.abspath(__file__))
```

Ruff (0.12.2)

`os.path.dirname()` should be replaced by `Path.parent`

(PTH120)